c7n/resources/aws.py

```diff
@@ -123,6 +125,56 @@ def shape_validate(params, shape_name, service):
         raise PolicyValidationError(report.generate_report())
 
 
+def get_bucket_region_clientless(bucket, s3_endpoint):
+    """Attempt to determine a bucket region without a client
+
+    We can make an unauthenticated HTTP HEAD request to S3 in an attempt to find a bucket's
+    region. This avoids some issues with cross-account/cross-region uses of the
+    GetBucketPolicy API action. Because bucket names are unique within
+    AWS partitions, we can make requests to a single regional S3 endpoint
+    and get redirected if a bucket lives in another region within the
+    same partition.
+
+    This approach is inspired by some sample code from a Go SDK issue comment,
+    which @sean-zou mentioned in #7593:
+
+    https://github.com/aws/aws-sdk-go/issues/720#issuecomment-613038544
+
+    Return a region string, or None if we're unable to determine one.
+    """
+    region = None
+    s3_endpoint_parts = urlparse(s3_endpoint)
+    bucket_endpoint = f'https://{bucket}.{s3_endpoint_parts.netloc}'
```

Comment:
this doesn't seem to work for buckets with a '.' in their name.

Reply:
True... possible workarounds I can see:

- Using HTTP rather than HTTPS
- Ignoring certificate validation
- Switching to a path-style URL (keeping https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/ in mind)

I'm leaning toward that last one. Thanks, good catch!